crates/ruff_linter/src/rules/pylint/rules/redefined_slots_in_subclass.rs

```diff
@@ -0,0 +1,204 @@
+///
+/// ## Example
+/// ```python
+/// class Base:
+///     __slots__ = ("a", "b")
+///
+///
+/// class Subclass(Base):
+///     __slots__ = ("a", "d")  # [redefined-slots-in-subclass]
+/// ```
+///
+/// Use instead:
+/// ```python
+/// class Base:
+///     __slots__ = ("a", "b")
+///
+///
+/// class Subclass(Base):
+///     __slots__ = "d"
+/// ```
+#[violation]
+pub struct RedefinedSlotsInSubclass {
+    name: String,
+}
+
+impl Violation for RedefinedSlotsInSubclass {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        let RedefinedSlotsInSubclass { name } = self;
+        format!("Redefined slots ['{name}'] in subclass")
+    }
+}
+
+// PLW0244
+pub(crate) fn redefined_slots_in_subclass(checker: &mut Checker, body: &[Stmt]) {
+    for slot in redefined_slots(body) {
+        checker.diagnostics.push(Diagnostic::new(
+            RedefinedSlotsInSubclass {
+                name: slot.name.to_string(),
+            },
+            slot.range(),
+        ));
+    }
+}
+
+#[derive(Clone, Hash, Debug, Eq, PartialEq)]
+struct Slot<'a> {
+    name: &'a str,
+    range: TextRange,
+}
+
+impl Ranged for Slot<'_> {
+    fn range(&self) -> TextRange {
+        self.range
+    }
+}
+
+fn redefined_slots(body: &[Stmt]) -> Vec<Slot> {
+    // First, collect all the slot attributes that are assigned to classes.
+    let mut class_slots = HashMap::new();
+    let mut class_bases = HashMap::new();
+    for stmt in body {
+        let Stmt::ClassDef(ast::StmtClassDef {
+            name,
+            body,
+            arguments,
+            ..
+        }) = stmt
+        else {
+            continue;
+        };
+        class_slots.insert(name.as_str(), slots_members(body));
+
+        if let Some(arguments) = arguments {
+            let mut bases = FxHashSet::default();
+            for base in arguments.args.iter() {
+                let Expr::Name(ast::ExprName { id, .. }) = base else {
+                    continue;
+                };
+                bases.insert(id.as_str());
+            }
+            class_bases.insert(name.as_str(), bases);
+        };
+    }
+
+    if class_slots.is_empty() || class_bases.is_empty() {
+        return vec![];
+    }
+
+    // Second, find redefined slots.
+    let mut slots_redefined = vec![];
+    for (klass, bases) in &class_bases {
+        if let Some(slots) = class_slots.get(klass) {
+            for base in bases {
+                if let Some(base_slots) = class_slots.get(base) {
+                    for slot in slots {
+                        for base_slot in base_slots {
+                            if slot.name == base_slot.name {
+                                slots_redefined.push(slot.to_owned());
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+    slots_redefined
+}
+
+fn slots_members(body: &[Stmt]) -> FxHashSet<Slot> {
```

Comment:
As mentioned earlier, I think we'd need to update this method to consider individual class and climb up the hierarchy to determine if there are any conflicts with any of the names in `__slots__`.